Desktop/data/expanddataproxymodel.h

```diff
@@ -3,18 +3,134 @@
 
 #include <QAbstractItemModel>
 #include "utils.h"
+#include <QUndoCommand>
+#include <json/json.h>
 
-class DataSetView;
+class ExpandDataProxyModel;
+
+class UndoModelCommand : public QUndoCommand
```

Comment:
Can these undo model classes be moved to their own header + implementation file? Not necessarely for each one a file but I think they need to be separated from the expanded proxy model nonetheless